### ISSUE TITLE
Fix the build system

### ISSUE THREAD

**Issue by prologic** [prologic/ulinux#28]
The build system is apparently broken and I'm currently unable to make a new release 😳￼

**Comment by markwylde** [comment]
Just came across this, wasn't sure if it was just mac issues. Trying on my linux now.

**Comment by markwylde** [comment]
Ahh this is just my own network issues. It appears to be downloading now. However, taking more than 5 minutes.

**Comment by markwylde** [comment]
There's a lot of dependencies out of date. I'm going through them and updating them as I find them, but each time I need to rebuild the linux kernel, which takes ages.

Can we utilise the Dockerfile to make it resume from the last successful line. I guess it's because the directory is changing so it thinks it has to start from the start.

**Comment by markwylde** [comment]
Okay, I see it's actually the Makefile that does this. I can just comment it out :P

**Comment by prologic** [comment]
Oh wow! Thanks for trying to help me fix this 🤗 Yes looks like we need to update a bunch of dependencies as I think somethings change from under us

The Docker image though once you've built the builder image should be pretty quick for subsequent builds after that￼￼￼￼

**Comment by prologic** [comment]
Feel free to submit a pull request if you manage to get the build fully working!￼ I think you've made way more progress than I have 🤣￼￼

**Comment by markwylde** [comment]
I'm getting close... Just tinycc to go.

**Comment by markwylde** [comment]
Hit a bit of a wall now:

```
Building builder ...
sha256:47a6d9bd9e5a744ec61a5f3ca0b99a7b95fd6f7dd3ebdcda4475cd511044057a
Building uLinux ...
Building pkgs ...                       
  Building pkg kernel ...               [ OK ]
  Building pkg musl ...                 [ OK ]
  Building pkg sinit ...                [ OK ]
  Building pkg make ...                 [ ERR ]
=> downloading http://ftpmirror.gnu.org/gnu/make/make-4.3.tar.lz
--2020-11-08 07:06:56--  http://ftpmirror.gnu.org/gnu/make/make-4.3.tar.lz
Resolving ftpmirror.gnu.org (ftpmirror.gnu.org)... failed: Name does not resolve.
wget: unable to resolve host address 'ftpmirror.gnu.org'
ERROR> failed to download http://ftpmirror.gnu.org/gnu/make/make-4.3.tar.lz
ls: cannot access '/build/artifacts/make#*.tar.gz': No such file or directory
ERROR> cannot figure out which package to add
FAIL:   Building pkg make failed
  Building pkg busybox ...              [ ERR ]
=> downloading https://busybox.net/downloads/snapshots/busybox-20201108.tar.bz2
--2020-11-08 07:06:57--  https://busybox.net/downloads/snapshots/busybox-20201108.tar.bz2
Resolving busybox.net (busybox.net)... failed: Name does not resolve.
wget: unable to resolve host address 'busybox.net'
ERROR> failed to download https://busybox.net/downloads/snapshots/busybox-20201108.tar.bz2
ls: cannot access '/build/artifacts/busybox#*.tar.gz': No such file or directory
ERROR> cannot figure out which package to add
FAIL:   Building pkg busybox failed
  Building pkg dropbear ...             [ ERR ]
=> downloading https://matt.ucc.asn.au/dropbear/dropbear-2020.81.tar.bz2
--2020-11-08 07:06:58--  https://matt.ucc.asn.au/dropbear/dropbear-2020.81.tar.bz2
Resolving matt.ucc.asn.au (matt.ucc.asn.au)... failed: Name does not resolve.
wget: unable to resolve host address 'matt.ucc.asn.au'
ERROR> failed to download https://matt.ucc.asn.au/dropbear/dropbear-2020.81.tar.bz2
ls: cannot access '/build/artifacts/dropbear#*.tar.gz': No such file or directory
ERROR> cannot figure out which package to add
FAIL:   Building pkg dropbear failed
  Building pkg extlinux ...             [ ERR ]
=> downloading http://kernel.org/pub/linux/utils/boot/syslinux/syslinux-6.03.tar.xz
--2020-11-08 07:06:58--  http://kernel.org/pub/linux/utils/boot/syslinux/syslinux-6.03.tar.xz
Resolving kernel.org (kernel.org)... failed: Name does not resolve.
wget: unable to resolve host address 'kernel.org'
ERROR> failed to download http://kernel.org/pub/linux/utils/boot/syslinux/syslinux-6.03.tar.xz
ls: cannot access '/build/artifacts/extlinux#*.tar.gz': No such file or directory
ERROR> cannot figure out which package to add
FAIL:   Building pkg extlinux failed
[ OK ]
FAIL: Build failed
Building ports ...                      
  Building port filesystem ...          [ OK ]
  Building port rc ...                  [ OK ]
  Building port svc ...                 [ OK ]
  Building port cloudinit ...           [ OK ]
  Building port net ...                 [ OK ]
  Building port services ...            [ OK ]
  Building port ca-certificates ...     [ OK ]
  Building port tcc ...                 [ ERR ]
=> downloading https://github.com/mirror/tinycc/archive/8b69059f66c0ec402d9e81815978e7f70b7181d8.zip
--2020-11-08 07:06:58--  https://github.com/mirror/tinycc/archive/8b69059f66c0ec402d9e81815978e7f70b7181d8.zip
Resolving github.com (github.com)... failed: Name does not resolve.
wget: unable to resolve host address 'github.com'
ERROR> failed to download https://github.com/mirror/tinycc/archive/8b69059f66c0ec402d9e81815978e7f70b7181d8.zip
ls: cannot access '/build/artifacts/tcc#*.tar.gz': No such file or directory
ERROR> cannot figure out which package to add
FAIL:   Building port tcc failed
  Building port pkg ...                 [ OK ]
  Building port setup ...               [ OK ]
  Building port box ...                 [ OK ]
[ OK ]
Building rootfs ...                     [ OK ]
Building iso ...                        [ OK ]
🎉 All Done!
Copying ISO Image(s), and RootFS ...
ulinux_build
Creating Disk Image ...
make: qemu-img: No such file or directory
make: *** [build] Error 1
```

It looks like the files are either not saving

**Comment by markwylde** [comment]
No. Network issues again. My docker is not playing nice today.

**Comment by prologic** [comment]
No worries I'll pick up where you left off t

**Comment by prologic** [comment]
I'm also getting random DNS resolution failures when building. Hmmm 🤔

**Comment by prologic** [comment]
Fixed with various commits on `master` and #31
